src/language/branches.md

```diff
@@ -0,0 +1,279 @@
+
+## Switch
+
+Switch statements evaluate a value expression against a series of cases.
+Each case clause is a [pattern][] followed by a series of statements in the case
+body to execute if the value matches that pattern. 
+
+Non-empty `case` clauses implicitly jump to the end of the switch after completion.
+Other valid ways to end a non-empty `case` clause are a [`continue`][break],
+[`throw`][], or [`return`][] statement.
+
+Use a `default` or wildcard `_` clause to execute code when no `case` clause matches:
+
+```dart
+var command = 'OPEN';
+switch (command) {
+  case 'CLOSED':
+    executeClosed();
+  case 'PENDING':
+    executePending();
+  case 'APPROVED':
+    executeApproved();
+  case 'DENIED':
+    executeDenied();
+  case 'OPEN':
+    executeOpen();
+  default:
+    executeUnknown();
+}
+```
+
+Empty cases fall through to the next case. For non-sequential fall-through,
+you can use a [`continue` statement][break] and a label:
+
+```dart
+switch (command) {
+  case 'OPEN':
+    executeOpen();
+    continue newCase;     // Continues executing at the newCase label.
+  
+  case 'DENIED':          // Empty case falls through.
+  case 'CLOSED':
+    executeClosed();      // Runs for both DENIED and CLOSED,
+  
+  newCase:
+  case 'PENDING':
+    executeNowClosed();   // Runs for both OPEN and PENDING.
+}
+```
+For an empty case that does not fall through, use [`break`][break] for its body.
+
+### Case patterns
+
+The pattern following a case, or case pattern, allows control flow to
+either match and destructure the object being switched on, or continue execution
+if the object doesn't match. Case patterns can be any kind of [pattern][]:
+
+- **Constant patterns**: the object's value has to be equivalent to match.
+  ```dart
+  switch (number) {
+    case 1:           // Matches if number == 1
+      print("one");     
+  }
+  ```
+- **List**, **map**, or **record patterns**: the elements are destructured if matched.
+  ```dart
+  switch (obj) {
+    case [a, b]:                  // Matches if obj is a list with two elements
+      print("$a and $b match");    
+  }
+  ```  
+- **Variable patterns**: the matched value binds to a new variable.
+  ```dart
+  switch ((1, 2)) {
+    case (var a, var b): ...     // Matches if 1 and 2 can be assigned to a and b
+  }
+  ```  
+- **Logical** or **relational patterns**: the object matches if the comparison to
+a constant returns `true`.
+  ```dart
+  switch (code) {
+    case >= first && <= last: ...    
+    // Matches if the value of code is between the constant values of first and last
+  }
+  ```  
+- **Null check** or **assert patterns**: the match is first on whether the object is not null, then on the value.
+  ```dart
+  switch (maybe) {
+    case var s?:                 
+    // Matches if maybe is not null, and then against the inner pattern "var s"
+  }
+  ```  
+- **Object patterns**: the match exposes getters on the object type.
```

Comment:
Maybe "Matches if the object has the given named type. If so, then calls getters on the object and matches them against corresponding inner patterns.

Reply:
Also moved and addressed this one, resolving!